Dockerfile

```diff
@@ -0,0 +1,21 @@
+# Use an official Python runtime as a parent image
+FROM python:3.11
+
+# Set the working directory in the container to /app
+WORKDIR /app
+
+# Add the current directory contents into the container at /app
+ADD . /app
+
+# Install any needed packages specified in requirements.txt
+RUN pip install --upgrade pip && \
+    pip install --extra-index-url https://packages.pathway.com/966431ef6ba -r requirements.txt
```

Comment:
```suggestion
COPY requirements.txt requirements.txt

# Install any needed packages specified in requirements.txt
RUN pip install --upgrade pip && \
    pip install --extra-index-url https://packages.pathway.com/966431ef6ba -r requirements.txt

COPY data data
COPY llm_app llm_app
```

1) using COPY instead of ADD
2) COPY requirements.txt first, run pip install and only then copy rest of the app. That way, upon changing code in the app, the cache for installed packages can be used
Reference: https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#add-or-copy